### PR TITLE
feat: adjust the fetching of the data from the firebase collection (D…

### LINKED ISSUE #22: Purchase History

### PR DESCRIPTION
**Pull request by walidbelharazem**
…onors)
adjust the fetching of the data from the firebase donors collection (with a fixed restaurant ID) and added code comments.
#22

# Related Issue

- Resolve #22

### CODE REVIEW

**Review by merzoukfatima — APPROVED** [review]
Great work walid starting to fetch from Firestore amazing 🙏

**Review by mouloud247 — APPROVED** [review]

File: src/components/Historycard.jsx [hunk 1]
```diff
@@ -1,72 +1,74 @@
...
 export default function Historycard() {
-    const [blogs, setBlogs] = useState([]);
-    const fetchBlogs = async () => {
-        const response = collection(db, "blogs");
-        const data = await getDocs(response);
-        data.docs.forEach((item) => {
-            setBlogs([...blogs, item.data()]);
-        });
-    };
+    const [blogData, setBlogData] = useState([]);
```
Comment (ikoworld) [inline src/components/Historycard.jsx:6]: why did you call it blogs ? i think is better to keep the names related to the data you are fetching as donors

File: src/components/Historycard.jsx [hunk 2]
```diff
@@ -1,72 +1,74 @@
...
+
     useEffect(() => {
+        async function fetchBlogs() {
```
Comment (ikoworld) [inline src/components/Historycard.jsx:9]: same thing here for the name of the function

File: src/components/Historycard.jsx [hunk 3]
```diff
@@ -1,72 +1,74 @@
...
+                    };
+                });
+                setBlogData(data);
+            } catch (error) {
+                console.error("Error fetching data:", error);
+            }
+        }
+
         fetchBlogs();
     }, []);
```
Comment (ikoworld) [inline src/components/Historycard.jsx:37]: well done

**Review by ikoworld — APPROVED** [review]
well done except the parts that i commented on the code is functional and i appreciate the comments that made it clear

**Pull request merged**